### PR TITLE
PageLayout: Update `divider` prop responsive API

### PR DESCRIPTION
**Pull request by colebemis**
* Updated the `divider` prop in `PageLayout.Header`, `PageLayout.Pane`, and `PageLayout.Footer` to use the new responsive prop API introduced in https://github.com/primer/react/pull/2174.
* Deprecated the `dividerWhenNarrow` prop in favor of the new responsive prop API

**Before**

```
divider="line"
dividerWhenNarrow="filled"
```

**After**

```
divider={{regular: 'line', narrow: 'filled'}}
```

### Merge checklist

- [ ] Added/updated tests
- [x] Added/updated documentation
- [x] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge

Take a look at the [What we look for in reviews](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#what-we-look-for-in-reviews) section of the contributing guidelines for more information on how we review PRs.

Part of https://github.com/github/primer/issues/793

### CODE REVIEW

File: src/PageLayout/PageLayout.tsx [hunk 1]
```diff
@@ -197,6 +197,13 @@ const Header: React.FC<React.PropsWithChildren<PageLayoutHeaderProps>> = ({
   children,
   sx = {}
 }) => {
+  // Combine divider and dividerWhenNarrow for backwards compatibility
```
Comment (siddharthkp) [inline src/PageLayout/PageLayout.tsx:200]: 💯

File: .changeset/spotty-parents-cheat.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+```
+divider="line"
+dividerWhenNarrow="filled"
+```
+
+**After**
+
+```
+divider={{regular: 'line', narrow: 'filled'}}
+```
```
Comment (siddharthkp) [inline .changeset/spotty-parents-cheat.md:19]: this will be really helpful for next major changelog 🏆

File: docs/content/PageLayout.mdx [hunk 3]
```diff
@@ -172,25 +173,39 @@ See [storybook](https://primer.style/react/storybook?path=/story/layout-pagelayo
...
+      | 'none'
+      | 'line' 
+      | 'filled'
+    regular?:
+      | 'none'
+      | 'line'
+    wide?:
+      | 'none'
+      | 'line'
+  }`}
```
Comment (siddharthkp) [inline docs/content/PageLayout.mdx:188]: (not blocker) we should think about how we want to show "type" in the docs, showing the typescript type isn't the most readable 😅

image

File: src/PageLayout/PageLayout.tsx [hunk 4]
```diff
@@ -197,6 +197,13 @@ const Header: React.FC<React.PropsWithChildren<PageLayoutHeaderProps>> = ({
...
+  const dividerProp =
+    !isResponsiveValue(divider) && dividerWhenNarrow !== 'inherit'
```
Comment (siddharthkp) [inline src/PageLayout/PageLayout.tsx:202]: oof! full points for backwards compatibility but i don't think we should combine the two APIs

use the recommended API if given, fallback to deprecated one if not. either new or old, both

**Review by siddharthkp — CHANGES_REQUESTED** [review]

**Review by siddharthkp — APPROVED** [review]

File: docs/content/PageLayout.mdx (hunk 3 above)
Comment (colebemis) [reply docs/content/PageLayout.mdx:188]: Agreed! Open to ideas here.

File: src/PageLayout/PageLayout.tsx (hunk 4 above)
Comment (colebemis) [reply src/PageLayout/PageLayout.tsx:202]: Yup. As soon as you use a pass a "responsive object" value, you're opting into the new API and we ignore the `*whenNarrow` prop 👍

**Pull request merged**